### PR TITLE
abdullah JS3 week2

### PR DESCRIPTION
**Pull request by heartwhite**
week1's index.js converted to promise version.
and made some changes based on Jim's last comments on week1

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -30,15 +32,116 @@
...
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  function makeHtmlOfContents(repos, index) {
+    document.querySelector('.leftDiv').innerHTML = '';
+    document.querySelector('.rightDiv').innerHTML = '';
+    makeTable(repos[index]);
+    fetchJSON(repos[index].contributors_url).then(data => makeContributorsList(data));
```
Comment (remarcmij) [inline homework/index.js]: You also need to catch and render errors here.

File: homework/index.js [hunk 2]
```diff
@@ -30,15 +32,116 @@
...
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.onclick = () => {
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  function makeHtmlOfContents(repos, index) {
```
Comment (remarcmij) [inline homework/index.js]: This function has no use for the complete array of repositories. It deals with just one repository (the selected one). In general, it is best to just give functions the exact data they need and nothing more (on a 'need to know' basis). This helps to make your code more robust as there is less chance of unintended data corruption. You could define this function like this:

```js
function  makeHtmlOfContents(selectedRepo) {
  ...
}
```

and call it like this:

```js
makeHtmlOfContents(repos[selectMenu.selectedIndex]);
```

File: homework/index.js [hunk 3]
```diff
@@ -30,15 +32,116 @@
     return elem;
   }
 
+  function makeHeader(repos) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('h3', header, { text: 'HYF Repositories' });
+    const selectMenu = createAndAppend('select', header, { class: 'repo-selector' });
+    const optionsArr = [];
```
Comment (remarcmij) [inline homework/index.js]: You can get rid of the complete `optionArr` argument: use the `repos` array directly:

```js
function makeHeader(repos) {
  const root = document.getElementById('root');
  const header = createAndAppend('header', root, { class: 'header' });
  createAndAppend('h3', header, { text: 'HYF Repositories' });
  const selectMenu = createAndAppend('select', header, { class: 'repo-selector' });
  repos.map((repo, index) =>
    createAndAppend('option', selectMenu, {
      value: index,
      text: repo.name,
    }),
  );
  return header;
}
```

File: homework/style.css [hunk 4]
```diff
@@ -1,3 +1,108 @@
 .alert-error {
```
Comment (remarcmij) [inline homework/style.css]: While your app is responsive, the select element needs some margin on the right:

image

File: homework/index.js (hunk 1 above)
Comment (heartwhite) [reply homework/index.js]: I added .catch

File: homework/index.js (hunk 2 above)
Comment (heartwhite) [reply homework/index.js]: yes I applied and .. I don't know why I wrote that much code before..

File: homework/style.css (hunk 4 above)
Comment (heartwhite) [reply homework/style.css]: I could figure out with 'min-width'

File: homework/index.js (hunk 3 above)
Comment (heartwhite) [reply homework/index.js]: I solved it but I forgot comment here

**Pull request closed without merging**